### ISSUE TITLE
Reason for getting different no. of detections and difference between IoU and confidence score 

### ISSUE THREAD

**Issue by hlmhlr** [fizyr/keras-retinanet#1518]
Hi, 

When i evaluate my csv dataset using evaluate.py function (available inside the bin folder with default IoU value of 0.5), i get total 13,807 no. of predictions for my validation set images. Now, when i run the predict.py function where confidence score is 0.5 with same validation data set, i get 10798 number of predictions.  predict.py function is same as "keras-retinanet/examples/ResNet50RetinaNet.ipynb(Run detection on example)" in the repository.  

I am wondering that why there is such difference in no. of predictions? What is difference between confidence score used in predict.py function and the IoU used in evaluate.py function. However, in both functions same function (boxes, scores, labels) = model.predict_on_batch(image) is used for the predictions. 

Consequently, when i calculate the AP metrics, i am getting different results. Can somebody clarify the above point. I would be really thankful. 

Kind Regards,
